README.md

```diff
@@ -1,2 +1,188 @@
-# spacelift-cli
-Spacelift client and CLI
+# Spacelift CLI
+
+Spacelift CLI (`spacelift-cli`) is a utility wrapping Spacelift's [GraphQL API](https://docs.spacelift.io/integrations/api) for easy programmatic access in command-line contexts - either in manual interactive mode (in your local shell), or in a predefined CI pipeline (GitHub actions, CircleCI, Jenkins etc).
+
+## Installation
+
+Spacelift CLI is distributed through GitHub Releases as a gzipped tarball containing a self-contained statically linked executable built from the source in this repository. Binaries can be download directly from the [Releases page](https://github.com/spacelift-io/spacelift-cli/releases) or programmatically using `curl` or `wget`. The example below covers macOS (Intel CPU) installation of the latest available release. In order to download and install a Linux version, change `darwin` to `linux`. If you're using an ARM processor (eg. Raspberry Pi, Apple Silicon), change `amd64` to `arm64`:
+
+```bash
+curl -s -L https://github.com/spacelift-io/spacelift-cli/releases/latest/download/spacelift-cli-darwin-amd64.tar.gz | tar xz -C /tmp && \
+mv /tmp/spacelift-cli-darwin-amd64 /usr/local/bin/spacelift-cli
+```
+
+## Authentication
+
+Spacelift CLI is designed to work in two different contexts - a non-interactive scripting mode (eg. external CI/CD pipeline) and local interactive mode, meaning you typing things in your shell. It thus supports two types of credentials - environment variables and user profiles. Let's call these "credential providers", like AWS does. We'll look into them separately.
+
+### Authenticating using environment variables
```

Comment:
@marcinwyszynski this is very subjective, so feel free to ignore me here. What you've written is fine - I just felt like it could be easier to quickly understand the options if we split things up a bit into a section for each authentication method, and were explicit in each section about exactly what variables are required.

Here's my suggested rewording:

```markdown
### Authenticating using environment variables

The spacelift CLI supports the following authentication methods via the environment:

- [Spacelift API tokens](#spacelift-api-tokens).
- [GitHub tokens](#github-tokens).
- [Spacelift API keys](#spacelift-api-keys).

The Spacelift CLI looks for authentication configurations in the order specified above,
and will stop as soon as it finds a valid configuration. For example, if a Spacelift API
token is specified, GitHub tokens and Spacelift API keys will be ignored, even if their
environment variables are specified.

#### Spacelift API tokens

Spacelift API tokens can be specified using the `SPACELIFT_API_TOKEN` environment variable.
When this variable is found, the CLI ignores all the other authentication environment variables
because the token contains all the information needed to authenticate.

NOTE: API tokens are generally short-lived and will need to be re-created often.

#### GitHub tokens

GitHub tokens are only available to accounts that use GitHub as their identity provider, but
are very convenient for use in GitHub actions. To use a GitHub token, set the following
environment variables:

- `SPACELIFT_API_ENDPOINT` - the URL to your Spacelift account, for example `https://mycorp.app.spacelift.io`.
- `SPACELIFT_API_GITHUB_TOKEN` - a GitHub personal access token.

#### Spacelift API keys

To use a Spacelift API key, set the following environment variables:

- `SPACELIFT_API_ENDPOINT` - the URL to your Spacelift account, for example `https://mycorp.app.spacelift.io`.
- `SPACELIFT_API_KEY_ID` - the ID of your Spacelift API key. Available via the Spacelift application.
- `SPACELIFT_API_KEY_SECRET` - the secret for your API key. Only available when the secret is created.

More information about API authentication can be found at <https://docs.spacelift.io/integrations/api#authenticating-with-the-api>.
```

Reply:
I understand what you mean but your version does not allow the reader to understand the order of precedence.

Reply:
Yeah - I tried to cover that with the following, but it's maybe not super clear:

> The Spacelift CLI looks for authentication configurations in the order specified above,
> and will stop as soon as it finds a valid configuration.

Anyhow - like I say it's really subjective 🤷

Reply:
I missed that entirely.

Reply:
Accepted the suggestion.